diff --git a/org.kde.keysmith.desktop b/org.kde.keysmith.desktop
index 24583a9..1c54940 100644
--- a/org.kde.keysmith.desktop
+++ b/org.kde.keysmith.desktop
@@ -85,10 +85,16 @@ Comment[uk]=Керування вашими ключами OTP для двофа
 Comment[zh_CN]=管理 OTP 令牌的双重身份验证 (2fa)
 Comment[zh_TW]=管理兩步驟驗證 (2FA) 的 OTP 憑證
 Version=1.0
-Exec=keysmith %u
+Exec=qt6-start.sh keysmith %u
 Icon=org.kde.keysmith
 Type=Application
 SingleMainWindow=true
 Terminal=false
 Categories=Qt;KDE;Utility
 MimeType=x-scheme-handler/otpauth;
+
+X-Nemo-Single-Instance=no
+X-Nemo-Application-Type=no-invoker
+
+[X-Sailjail]
+Sandboxing=Disabled
